src/main/java/seedu/address/model/student/Year.java

```diff
@@ -25,17 +26,26 @@
 
     /**
      * Constructs a {@code Year}.
-     *
-     * @param schoolType A valid school type
-     * @param level A valid level
      */
-    public Year(SchoolType schoolType, Integer level) {
-        requireNonNull(schoolType);
-        requireNonNull(level);
+    public Year(String year) {
+        requireNonNull(year);
+        final Matcher matcher = YEAR_FORMAT.matcher(year.trim());
+        assert matcher.matches();
```

Comment:
For this line I think just take note because `matcher.matches()` actually changes the state of matcher, when assertions are not used there might be bugs.

Reply:
Good catch, I'll change it and also include a `checkArgument()` call in the constructor.